src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,134 @@
+                () -> generateElevatorMotorProfile(targetPosition),
+                this::setTargetPositionFromProfile,
+                (interrupted) -> {
+                },
+                () -> false,
+                this
+        );
+    }
+
+    /**
+     * @return a command that creates the profile for the angle motor
+     */
+    private void generateAngleMotorProfile(Rotation2d targetAngle) {
+        angleMotorProfile = new TrapezoidProfile(
+                ArmConstants.ANGLE_CONSTRAINTS,
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(getAngleMotorPositionDegrees(), getAngleMotorVelocity())
+        );
+        lastAngleMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    /**
+     * @return a command that creates the profile for the elevator motor
+     */
+    private void generateElevatorMotorProfile(double targetPosition) {
+        elevatorMotorProfile = new TrapezoidProfile(
+                ArmConstants.ELEVATOR_CONSTRAINTS,
+                new TrapezoidProfile.State(targetPosition * ArmConstants.ELEVATOR_MOTOR_RATIO, 0),
+                new TrapezoidProfile.State(getElevatorMotorPositionDegrees(), getElevatorMotorVelocity())
+        );
+        lastElevatorMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    /**
+     * @return a command that moves the angle motor to the target angle
+     */
+    private void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            masterAngleMotor.stopMotor();
+            return;
+        }
+        final TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleMotorProfileTime());
+        masterAngleMotor.getPIDController().setReference(targetState.position, CANSparkMax.ControlType.kPosition);
+    }
+
+    /**
+     * @return a command that moves the elevator motor to the target position
+     */
+    private void setTargetPositionFromProfile() {
+        if (elevatorMotorProfile == null) {
+            masterElevatorMotor.stopMotor();
+            return;
+        }
+        final TrapezoidProfile.State targetState = elevatorMotorProfile.calculate(getElevatorMotorProfileTime());
+        masterElevatorMotor.getPIDController().setReference(targetState.position, CANSparkMax.ControlType.kPosition);
+    }
+
+    private double getAngleMotorProfileTime() {
+        return Timer.getFPGATimestamp() - lastAngleMotorProfileGenerationTime;
+    }
+
+    private double getElevatorMotorProfileTime() {
+        return Timer.getFPGATimestamp() - lastElevatorMotorProfileGenerationTime;
+    }
+
+    private double getAngleMotorPositionDegrees() {
+        return angleEncoder.getAbsolutePosition().getValue();
+    }
+
+    private double getElevatorMotorPositionDegrees() {
+        return elevatorEncoder.getSelectedSensorPosition();
+    }
+
+    private double getAngleMotorVelocity() {
+        return angleEncoder.getVelocity().getValue();
+    }
```

Comment:
Either append `DegreesPerSecond` to the name of this function or make this return a Rotation2d